Document custom controls in style guide and how to instance them (#3470)
Revolutionary-Games/Thrive: doc/style_guide.md

- You should follow general GUI standards in designing UI. Use widgets
  that are meant for whatever kind of interaction you are designing.

> - We have rewritten several controls to workaround Godot bugs or limits, and

Reviewer: ```suggestion
- We have rewritten several controls to workaround Godot bugs or limitations, and
```

>   add custom features. All these rewritten controls are placed in

Reviewer: ```suggestion
  add custom features. All these rewritten/customized controls are placed in
```

  "res://src/gui_common/". Currently there are `CustomCheckBox`,
  `CustomDialog`,  `CustomConfirmationDialog`, `ErrorDialog`,
  `TutorialDialog`, `CustomDropDown`, `CustomRichTextLabel`, and

>   `TweakedColourPicker`. When adding control types listed above, consider using
>   the custom types rather than the built-in types to ensure consistency
>   across the GUI. This is because the custom implementation offer a much more
>   customized styling and additional functionality.

Reviewer: ```suggestion
  `TweakedColourPicker`. Consider using these custom types rather than the
  built-in types to ensure consistency across the GUI.
```
The last sentence is redundant as it's already somewhat mentioned in the opening sentence.

> - When you are instancing a custom Control in code, use

Reviewer: ```suggestion
- When you are instantiating a custom Control in code, use
```
`Instantiating` is actually the more correct usage of the verb and `Instance` will be changed into `Instantiate` in 4.0 anyway.

Author: Emm ...

>   `(T)GD.load<PackedScene>("res://src/gui_common/T.tscn").Instance()` if the

Reviewer: We usually have this done in two liner:
```csharp
var scene = GD.Load<PackedScene>("res://src/gui_common/T.tscn");
var instance = scene.Instance<T>();
```

Author: That's right, but I don't really want to bother. I'll change this anyway.

>   custom control has a .tscn scene; use `new T` if it doesn't.

Reviewer: ```suggestion
  custom control has a corresponding scene (.tscn) file; use `new T` if it
  doesn't.
```

- You should follow general GUI standards in designing UI. Use widgets
  that are meant for whatever kind of interaction you are designing.

- We have rewritten several controls to workaround Godot bugs or limitations,
  and add custom features. All these rewritten/customized controls are placed
  in "res://src/gui_common/". Currently there are `CustomCheckBox`,
  `CustomDialog`,  `CustomConfirmationDialog`, `ErrorDialog`,
  `TutorialDialog`, `CustomDropDown`, `CustomRichTextLabel`, and
  `TweakedColourPicker`. Consider using these custom types rather than the
  built-in types to ensure consistency across the GUI.

- When you are instantiating a custom Control in Godot, use

>   `Instance Child Scene` if the custom control has a corresponding scene

Reviewer: Could be good mentioning drag'n'drop.

>   `Instance Child Scene` if the custom control has a corresponding scene

Reviewer: Also for less verbosity:
```suggestion
  `Instance Child Scene` if it has a corresponding scene
```

  (.tscn) file; use `Attach Script` if it doesn't.

> - When you are instantiating a custom Control in code, use

Reviewer: ```suggestion
- When you are instantiating a custom Control in code, use the following
  if it has a corresponding scene (.tscn) file; use `new T` if it doesn't.
```
Whole sentence is more easier to understand.

- You should follow general GUI standards in designing UI. Use widgets
  that are meant for whatever kind of interaction you are designing.

- We have rewritten several controls to workaround Godot bugs or limitations,
  and add custom features. All these rewritten/customized controls are placed
  in "res://src/gui_common/". Currently there are `CustomCheckBox`,

>   `CustomDialog`,  `CustomConfirmationDialog`, `ErrorDialog`,

Reviewer 2: ```suggestion
  `CustomDialog`, `CustomConfirmationDialog`, `ErrorDialog`,
```
